### ISSUE TITLE
Creatures with darkvision have darkvision but dont have detection mode range

### ISSUE THREAD

**Issue by Aethen22** [MrPrimate/ddb-importer#292]
When you import a monster with darkvision(Such as Aboleth) the monster has darkvision(as in, it can see the map), but it doesnt detect tokens when there's no light(and no unrestricted vision range), thus making it see the map, but not tokens, my wild guess is that as it works with other types of visions, it searches for darkvision on the detection modes and as it doesnt find it, it creates the basic one with 0 range.

1. Import a monster with darkvision
2. Create the token in a scene where unrestricted vision range is off and its dark
3. Create a new token
4. Check how the monster cant see the token, as it cries thinking its alone.

**Comment by MrPrimate** [comment]
I can't replicate this ...  could you reach out on Discord to provide an export of your aboleth and scene please.

It's worth noting there is _no_ detection mode for darkvision in core 5e - so if you are seeing a detection mode for darkvision it maybe that a mod has added this.

The Vision Mode on the basic config tab should be set to "Darkvision" which will provide the basic darkvision mode.

The detection modes add other forms of detection, like seeing invisible things, or limiting things like basic sight to specific radius

**Comment by MrPrimate** [comment]
It might also be that you have your default token settings set up to provide a basic vision of 0, which will mean that the system won't generate a basic vision based on the tokens darkvision setting

**Comment by Vhostym-the-Sojourner** [comment]
I am having this same issue. What are the recommended default settings and why would the default be 0?

**Comment by MrPrimate** [comment]
If you've got vision enabled for the default tokens, then make sure there is no Detection Modes added.

**Comment by Vhostym-the-Sojourner** [comment]
Thanks! I think that may have been it. I haven't messed with that setting to my knowledge, but I did have an empty entry in Detection Modes that may have been causing it.

**Comment by Vhostym-the-Sojourner** [comment]
I tried to reimport the character after updating the setting you mentioned. It looks like even though Darkvision is specified on the import and it correctly imports to the character sheet, it does not update the default values for that actor's token. An imported character with 60 feet of darkvision still has their base vision selected and a range of 0.

**Comment by MrPrimate** [comment]
> I tried to reimport the character after updating the setting you mentioned. It looks like even though Darkvision is specified on the import and it correctly imports to the character sheet, it does not update the default values for that actor's token. An imported character with 60 feet of darkvision still has their base vision selected and a range of 0.

Character imports are different to monsters. When you import/update a character it does only update the prototype token, not the token on the canvas, so if the character's vision has changed you will need to delete the current token and drag a new one from the sidebar.

**Comment by Vhostym-the-Sojourner** [comment]
Absolutely. I am removing all tokens before the import, freshly importing characters, and then adding the tokens back and their vision is not being set.

**Comment by MrPrimate** [comment]
Okay, a bit stumped - if you could reach out on Discord with a login for your world, or do a screen share so I can take a look it would be helpful.

**Comment by notfelx** [comment]
Hello all,

Just faced same issue with importer setting a detection range with "blank" value to all imported tokens.
In my case, it was because the default token in "Settings > Core > Default Token Configuration" was set the same, with a blank detection mode.
![image](https://user-images.githubusercontent.com/15716813/230959163-c2aedecc-3de1-43d4-949d-d2607a7f9982.png)

If you reset detection modes
![image](https://user-images.githubusercontent.com/15716813/230958438-fb443f03-fd66-4c1d-bb81-8f6b2b3c924a.png)

Then click "enable vision"
![image](https://user-images.githubusercontent.com/15716813/230958512-6e96cf82-268c-49da-a18d-b3039fdc8f80.png)

and save.

If you re-open, you will see this 
![image](https://user-images.githubusercontent.com/15716813/230958616-035fc1dd-14e3-4502-8d09-f36fc4e3e1de.png)

This makes any new token (even without DDB Importer / manually !!) have a detection of 0, causing the bug.

This happens with no modules enabled on Foundry 10 / 291.
![image](https://user-images.githubusercontent.com/15716813/230959760-fdd9d8e5-1c71-4b2b-8f57-60fffad2e501.png)

anywhere I should raise this issue ?
